pkg/tenant/status_test.go

```diff
@@ -59,15 +53,17 @@ func TestAvailableMaintenance(t *testing.T) {
 	srv, err := tenant.New(conf)
 	require.NoError(t, err, "could not create tenant server in maintenance mode")
 
+	stopped := make(chan bool)
+	t.Cleanup(func() {
+		srv.Shutdown()
+		<-stopped
+		logger.ResetLogger()
+	})
```

Comment:
Weird deadlock that only occurred when this test errored out. Previously the `<-stopped` was declared in the cleanup earlier in the method which caused a hang because the channel was never written to if the test errored.